configure.ac

```diff
@@ -74,6 +74,12 @@ fi
 AM_CONDITIONAL(ENABLE_TESTS_GLES,
     [test "x$enable_tests_gles" = "xyes"])
 
+HAVE_GTEST="no"
+ifdef([GTEST_LIB_CHECK],
```

Comment:
I've worked on projects where the unit tests were defined directly within the same source files that were being tested.  But that can get messy real quick and requires extra care (e.g. with enable/disable flags, macros, and such) to ensure they don't pollute and add runtime overhead in the final product.

As long as the unit tests are defined in separate source files and compiled into their own binaries, there will be no additional runtime overhead or pollution in the libyami libraries, regardless of enabling gtest or not.  If you examine the makefile changes here and my WIP unit test branches (linked in the pull request summary), you'll notice there is a clear separation of the unit tests from the libyami libraries.